_posts/2021-12-31-trino-2021-a-year-of-growth.md

```diff
@@ -0,0 +1,420 @@
+
+## Data Sources
+
+### BigQuery connector
+
+* Added `CREATE TABLE` and `DROP TABLE` support.
+* Added support for case insensitive name matching for BigQuery views.
+* Support reading `bignumeric` type whose precision is less than or equal to 
+38.
+* Added support for `CREATE SCHEMA` and `DROP SCHEMA` statements.
+* Improved support for BigQuery datetime and timestamp types.
+
+### Cassandra connector
+
+* Mapped Cassandra `uuid` type to Trino `uuid`.
+* Added support for Cassandra `tuple` type.
+* Changed minimum number of speculative executions from two to one.
+* Support for reading user-defined types.
+
+### Clickhouse connector
+
+* Added [ClickHouse connector]({{site.url}}/docs/current/connector/clickhouse.html).
+* Improved performance of aggregation queries by computing aggregations within 
+ClickHouse. Currently, the following aggregate functions are eligible for
+pushdown: `count`, `min`, `max`, `sum` and `avg`.
+* Added support for dropping columns.
+* Map ClickHouse `UUID` columns as `UUID` type in Trino instead of `VARCHAR`.
+
+### HDFS, S3, Azure and cloud object storage systems
+
+A core use case of Trino uses the Hive and Iceberg connectors to connect to
+a data lake. These connectors differ from most as Trino is the sole query engine
+as opposed to the client calling another system. Here are some changes that
+for these connectors:
+
+* Enabled Glue statistics to support better query planning when using AWS.
+* `UPDATE` support for ACID tables
+* A lot of Hive view improvements.
+* Parquet column indexes.
+* Target file size feature (`target_max_file_size`).
+* Streaming uploads to S3 by default to improve performance and reduce disk usage.
+* Improved performance for tables with small files and partitioned tables.
+* Hive to Iceberg redirect.
+* Updated to Iceberg 0.11.0 behavior for transforms of dates and timestamps before 1970.
+* Added procedure `system.flush_metadata_cache()` to flush metadata caches.
+* Avoid generating splits for empty files.
+* Sped up Iceberg query performance when dynamic filtering can be leveraged.
+* Increased Iceberg performance when reading timestamps from Parquet files.
+* Improved Iceberg performance for queries on nested data through dereference pushdown.
+* Added support for `INSERT OVERWRITE` operations on S3-backed tables.
+* Made the Iceberg `uuid` type available.
+* Trino views made available in Iceberg.
+
+
+### Elasticsearch connector
+
+* Added support for reading fields as `json` values.
+* Fixed failure when documents contain fields of unsupported types.
+* Added support for `scaled_float` type.
+* Added support for assuming an IAM role.
+* Added retry requests with backoff when Elasticsearch is overloaded.
+* Better support for Elastic Cloud.
+
+### MemSQL connector
+
+* Allowed limiting the size of the metadata cache.
+* Added partial support for `DELETE` statement where predicate can be fully pushed 
+down to the remote datasource.
```

Comment:
I'll still say to add a section/statement like below (and remove the MemSQL section since just a version update is not newsworthy IMO):

```
In addition to the above some more features were added that apply to most JDBC based connectors:
...
```

Reply:
"JDBC based" is an internal implementation detail, and not something most people are familiar with. We'd need to either enumerate all the connectors or just say "to various connectors" and link to release notes or PRs for anyone who wants to get more details.